### ISSUE TITLE
RBD should also have support to be used directly from qemu

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#431]
Originally reported of Google Code with ID 375.

```
Optionally (perhaps via a disk parameter) it should be possible at least for kvm instances to skip activating rbd devices in the host's kernel, and to use them directly from qemu's userspace, for increased performance.

thanks,

Guido
```
Originally added on 2013-02-14 13:54:58 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is Pulkit's summer of code, so assigning for now to Thomas.
```
Originally added on 2013-07-26 12:49:31 +0000 UTC.

Changed State: Accepted
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Did this code get successfully completed in GSoC? Does somebody have a link to review it?
```
Originally added on 2013-09-21 08:04:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately, the GSoC project wasn't completed successfully. We are, however, actively working on the issue and are expecting support for this in Ganeti 2.10.
```
Originally added on 2013-09-23 06:21:42 +0000 UTC.

Added Labels: Priority-Medium Component-storage Hypervisor-KVM

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-09-24 10:55:09 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Usermode access has been written (commits 9b9e088..cbb533f) and should be released in Ganeti 2.10.
```
Originally added on 2013-10-19 08:54:32 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
2.10.0 with fixes for these bugs has just been released.
```
Originally added on 2014-02-20 13:25:23 +0000 UTC.

Changed State: Released
